File: OC/CustomActionSheet/ViewController.m
//
//  ViewController.m
//  CustomActionSheet
//
//  Created by RAIN on 15/12/7.
//  Copyright © 2015年 Smartech. All rights reserved.
//

#import "ViewController.h"
#import "CustomActionSheet.h"

@interface ViewController () <CustomActionSheetDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)buttonAction:(UIButton *)sender {
    NSArray * buttonArray = @[[CustomActionSheetButton buttonWithImage:[UIImage imageNamed:@"menu_add"] title:@"1111"],
                              [CustomActionSheetButton buttonWithImage:[UIImage imageNamed:@"menu_add"] title:@"2222"],
                              [CustomActionSheetButton buttonWithImage:[UIImage imageNamed:@"menu_add"] title:@"3333"],
                              [CustomActionSheetButton buttonWithImage:[UIImage imageNamed:@"menu_add"] title:@"4444"],
                              [CustomActionSheetButton buttonWithImage:[UIImage imageNamed:@"menu_add"] title:@"5555"]];
    
    CustomActionSheet * sheet = [[CustomActionSheet alloc] initWithButtons:buttonArray];
    sheet.delegate = self;
    [sheet showInView:self.view];
}

#pragma mark - CustomActionSheetDelegate
- (void)customActionSheet:(CustomActionSheet *)actionSheet
     clickedButtonAtIndex:(NSInteger)index
{
    NSLog(@"%@", @(index));
}

- (void)customActionSheetCancel:(CustomActionSheet *)actionSheet {
    NSLog(@"only dismiss");
}

@end

File: Swift/CustomActionSheet/ViewController.swift
//
//  ViewController.swift
//  CustomActionSheet
//
//  Created by RAIN on 15/11/30.
//  Copyright © 2015年 Smartech. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func buttonAction() {
        let sheet = CustomActionSheet()
        
        sheet.delegate = self
        
        sheet.setupViews([CustomActionSheetButton.buttonWithImage(UIImage(named: "menu_add")!, title: "1"),
            CustomActionSheetButton.buttonWithImage(UIImage(named: "menu_add")!, title: "2"),
            CustomActionSheetButton.buttonWithImage(UIImage(named: "menu_add")!, title: "3"),
            CustomActionSheetButton.buttonWithImage(UIImage(named: "menu_add")!, title: "4")])

        sheet.showInView(view)
    }

}

extension ViewController: CustomActionSheetDelegate {
    func customActionSheet(actionSheet: CustomActionSheet, choseAtIndex index: Int) {
        print(index)
    }
}

